src/ostree.cc

```diff
@@ -16,22 +16,22 @@
 #include "utils.h"
 
 data::InstallOutcome OstreeManager::pull(const Config &config, const KeyManager &keys, const std::string &refhash) {
-  OstreeRepo *repo = NULL;
   const char *const commit_ids[] = {refhash.c_str()};
   GCancellable *cancellable = NULL;
   GError *error = NULL;
   GVariantBuilder builder;
   GVariant *options;
 
   boost::shared_ptr<OstreeSysroot> sysroot = OstreeManager::LoadSysroot(config.pacman.sysroot);
-  if (!ostree_sysroot_get_repo(sysroot.get(), &repo, cancellable, &error)) {
+  boost::shared_ptr<OstreeRepo> repo = LoadRepo(sysroot, &error);
+  if (error) {
     LOG_ERROR << "Could not get OSTree repo";
     g_error_free(error);
     return data::InstallOutcome(data::INSTALL_FAILED, "Could not get OSTree repo");
   }
 
   GHashTable *ref_list = NULL;
-  if (ostree_repo_list_commit_objects_starting_with(repo, refhash.c_str(), &ref_list, nullptr, &error)) {
+  if (ostree_repo_list_commit_objects_starting_with(repo.get(), refhash.c_str(), &ref_list, nullptr, &error)) {
     guint length = g_hash_table_size(ref_list);
     g_hash_table_destroy(ref_list);  // OSTree creates the table with destroy notifiers, so no memory leaks expected
     // should never be greater than 1, but use >= for robustness
```

Comment:
The leak was here?

Reply:
It was just that not all calls to `ostree_sysroot_get_repo` were matched with `g_object_unref` on the acquired OstreeRepo. If you see more leaks here, please shout.